client/homebrew/editor/tagInput/tagInput.jsx

```diff
@@ -0,0 +1,105 @@
+require('./tagInput.less');
+const React = require('react');
+const { useState, useEffect } = React;
+const _     = require('lodash');
+
+const TagInput = ({ unique = true, values = [], ...props }) => {
+	const [temporaryValue, setTemporaryValue] = useState('');
+	const [valueContext, setValueContext] = useState(values.map((value) => ({ value, editing: false })));
+
+	useEffect(()=>{
+		handleChange(valueContext.map((context)=>context.value))
+	}, [valueContext])
+
+	const handleChange = (value)=>{
+		props.onChange({
+			target : { value }
+		})
+	};
+
+	const handleInputKeyDown = ({ evt, value, index, options = {} }) => {
```

Comment:
Any special reason this function takes an object instead of individual parameters?

Reply:
Not at computer right now but I believe it’s because it’s not always called with every parameter, so having  named properties helps